### PR TITLE
Add `preserveConsecutiveUppercase` option to `CamelCase` and add `SplitWords` type

### LINKED ISSUE #224: Feature Request: CamelCase to behave like lodash.camelCase and camelcase
Right now, `CamelCase` (and, obviously, `CamelCasedProperties` and `CamelCasedPropertiesDeep`) behaves weirdly when handling two character strings.

When it's a 2 character string (like `ID`, for example), it will transform the string as expected (`ID` to `id`), but strings like `HostIDPlayer` will leave the `ID` part untouched (`HostIDPlayer` to `hostIDPlayer`).

```ts
const a: CamelCase<'ID'> = 'id';  // Works, no TS error 
const b: CamelCase<'HostIDPlayer'> = 'hostIdPlayer';  // Error, TS expects 'hostIDPlayer'
```

Libraries like `lodash.camelCase` and `camelcase` will transform the strings as expected as can be seen in [this codesandbox](https://codesandbox.io/s/festive-leavitt-byu6y?file=/src/index.js).

I can't say I know enough about TS' type system, but is there a way to make `CamelCase` behave like `lodash.camelCase` and `camelcase`? That was, to me, what I expected the behavior to be.

### LINKED ISSUE #488: `CamelCase` does not properly camelize consecutive characters
Hi!

I'm trying to use the `CamelCase` type in a project. I have the following code:

```ts
type Sample = CamelCase<"GameID">;
```

My expectation according to the docs is `Sample` would now be typed as a string value of `"gameId"`. However, it is instead typed as `"gameID"`, with __ID__ being all uppercase.

This caught me off-guard, as the example in _camel-case.d.ts_ has this code listing:

```ts
interface RawOptions {
	'dry-run': boolean;
	'full_family_name': string;
	foo: number;
	BAR: string;
	QUZ_QUX: number; // <-- here
	'OTHER-FIELD': boolean;
}

const dbResult: CamelCasedProperties<RawOptions> = {
	dryRun: true,
	fullFamilyName: 'bar.js',
	foo: 123,
	bar: 'foo',
	quzQux: 6, // <-- here
	otherField: false
};
```

For my use case, this issue impacts the `CamelCasedProperties` and `CamelCasedPropertiesDeep` types, which leverage `CamelCase`.

### PR DESCRIPTION
**Pull request by skarab42**
Fix #224
Fix #488 

```ts
expectType<CamelCase<'fooBAR'>>('fooBAR');
expectType<CamelCase<'fooBAR', {preserveConsecutiveUppercase: false}>>('fooBar');

expectType<CamelCase<'fooBARBiz'>>('fooBARBiz');
expectType<CamelCase<'fooBARBiz', {preserveConsecutiveUppercase: false}>>('fooBarBiz');

expectType<CamelCase<'foo BAR-Biz_BUZZ'>>('fooBARBizBUZZ');
expectType<CamelCase<'foo BAR-Biz_BUZZ', {preserveConsecutiveUppercase: false}>>('fooBarBizBuzz');
```

In anticipation for #336 this PR also adds the `SplitWords` type (internal for now).

```ts
type Words0 = SplitWords<'helloWorld'>; // ['hello', 'World']
type Words1 = SplitWords<'helloWORLD'>; // ['hello', 'WORLD']
type Words2 = SplitWords<'hello-world'>; // ['hello', 'world']
type Words3 = SplitWords<'--hello the_world'>; // ['hello', 'the', 'world']
type Words4 = SplitWords<'lifeIs42'>; // ['life', 'Is', '42']
```

### CODE REVIEW

File: source/split-words.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import type {IsLowerCase, IsNumeric, IsUpperCase, WordSeparators} from './internal';
+
+type SkipEmptyWord<Word extends string> = Word extends '' ? [] : [Word];
+
+type RemoveLastChar<Sentence extends string, Char extends string> = Sentence extends `${infer LeftSide}${Char}`
```
Comment (sindresorhus) [inline source/split-words.d.ts:5]: `Char` => `Character`. Same for the other cases.

**Comment by sindresorhus** [comment]
Nice work 👍

**Comment by RichiCoder1** [comment]
It seems like this PR and `camelcase` have two different behaviors:
* `camelCase("TPSCode", { preserveConsecutiveUppercase: true})` => `TPSCode`
* `CamelCase<'TPSCode', {preserveConsecutiveUppercase: true}>` => `tPSCode`

**Comment by skarab42** [comment]
> Is there anyway to fix or consolidate this behavior?

@RichiCoder1  As a temporary fix you can use [Capitalize](https://www.typescriptlang.org/docs/handbook/utility-types.html#capitalizestringtype)

**Comment by ulken** [comment]
Since `PascalCase` [uses](https://github.com/sindresorhus/type-fest/blob/7294cf0e14883724f899ee2531cfed21fc46656e/source/pascal-case.d.ts#L1) `CamelCase` under the hood, I'd expect the options to propagate. Upgrading broke existing usages of `PascalCase`, without (seemingly) any way to get the old behavior back.

I can work around it, but I still think it should be configurable.

**Pull request merged**